gradle-plugin/src/main/java/com/microsoft/gradle/GradleProjectModelBuilder.java

```diff
@@ -93,6 +101,59 @@ public Object buildAll(String modelName, Project project) {
 		return rootModel;
 	}
 
+	private GradleProjectModel buildModel(Project rootProject, Project project) {
+		if (rootProject == null || project == null) {
+			return null;
+		}
+
+		ScriptHandler buildScript = project.getBuildscript();
+
+		ClassPath classpath = ((DefaultScriptHandler) buildScript).getScriptClassPath();
+
+		List<String> scriptClasspaths = classpath.getAsFiles().stream().map(file -> file.getAbsolutePath())
+				.collect(Collectors.toList());
+
+		GradleDependencyNode node = generateDefaultGradleDependencyNode(project);
+
+		List<String> plugins = getPlugins(project);
+
+		List<GradleClosure> closures = getPluginClosures(project);
+
+		Map<String, Project> childProjects = project.getChildProjects();
+
+		List<GradleProjectModel> subModels = childProjects.values().stream()
+				.map(childProject -> buildModel(rootProject, childProject)).filter(subModel -> subModel != null)
+				.collect(Collectors.toList());
+
+		List<GradleTask> tasks = getGradleTasks(rootProject, project);
+
+		return new DefaultGradleProjectModel(project.getParent() == null, project.getProjectDir().getAbsolutePath(),
+				subModels, tasks, node, plugins, closures, scriptClasspaths);
+	}
+
+	private List<GradleTask> getGradleTasks(Project rootProject, Project project) {
```

Comment:
Can we just directly fetch the tasks with this approach no matter the project's gradle version is > or < 5? Since they look very similar.

// The original code is not implemented by me so I'm not aware of the history of it.

Reply:
Sorry for long response, as far as I understand plugin's local gradle project model does not support some lazy tasks, so this commit was made as fix https://github.com/microsoft/vscode-gradle/pull/1302 in which now plugin uses gradle built in model. I couldn't reproduce this bug though so I can try to unify these methods to be compatible with everything, but it surely will need more testing